### PR TITLE
Add option to exclude keys

### PR DESCRIPTION
**Pull request by egoist**
As discussed in #4 😅

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,8 +2,9 @@
 var mapObj = require('map-obj');
 var camelCase = require('camelcase');
 
-module.exports = function (obj) {
+module.exports = function (obj, exclude) {
```
Comment (sindresorhus) [inline index.js]: Should be an options object, in case more options are needed in the future.

File: index.js [hunk 2]
```diff
@@ -2,8 +2,9 @@
...
+	exclude = exclude || [];
 	return mapObj(obj, function (key, val) {
-		return [camelCase(key), val];
+		return [exclude.indexOf(key) > -1 ? key : camelCase(key), val];
```
Comment (sindresorhus) [inline index.js]: Move `exclude.indexOf(key) > -1 ? key : camelCase(key)` into variable.

Comment (sindresorhus) [inline index.js]: `exclude.indexOf(key) > -1` => `exclude.indexOf(key) !== -1`

File: readme.md [hunk 3]
```diff
@@ -27,6 +27,24 @@ camelcaseKeys(argv);
...
+Type: `Object`
+
+Object to camelCase.
+
+#### exclude
+
+Type: `Array`  
+Default: `[]`
+
+Exclude keys from camelCasing.
```
Comment (sindresorhus) [inline readme.md]: > Exclude keys from being camelCased.

**Comment by egoist** [comment]
Updated 😁

File: index.js [hunk 4]
```diff
@@ -2,8 +2,11 @@
 var mapObj = require('map-obj');
 var camelCase = require('camelcase');
 
-module.exports = function (obj) {
+module.exports = function (obj, options) {
+	options = options || {};
+	options.exclude = options.exclude || [];
```
Comment (sindresorhus) [inline index.js]: `var exclude = options.exclude || []` and use the variable.

Right now you're modifying the user-supplied object, which is bad and will change the behavior of consecutive calls. Remember that objects are reference based.

File: index.js [hunk 5]
```diff
@@ -2,8 +2,11 @@
...
 	return mapObj(obj, function (key, val) {
-		return [camelCase(key), val];
+		var k = options.exclude.indexOf(key) === -1 ? camelCase(key) : key;
```
Comment (sindresorhus) [inline index.js]: Actually, just overwrite `key`.

```js
key = options.exclude.indexOf(key) === -1 ? camelCase(key) : key;
```

File: readme.md [hunk 6]
```diff
@@ -27,6 +27,28 @@ camelcaseKeys(argv);
 ```
 
 
+## API
+
+### camelcaseKeys(obj, [options])
+
+#### obj
```
Comment (sindresorhus) [inline readme.md]: `obj` => `input`

**Comment by egoist** [comment]
Thanks for the patience 👍🏻

**Comment by sindresorhus** [comment]
No, thank you :)

**Pull request closed without merging**